nova_vm/src/ecmascript/syntax_directed_operations/function_definitions.rs

```diff
@@ -251,3 +269,50 @@ pub(crate) fn evaluate_async_function_body(
     // 5. Return Completion Record { [[Type]]: return, [[Value]]: promiseCapability.[[Promise]], [[Target]]: empty }.
     promise_capability.promise()
 }
+
+/// ### [15.5.2 Runtime Semantics: EvaluateGeneratorBody](https://tc39.es/ecma262/#sec-runtime-semantics-evaluategeneratorbody)
+/// The syntax-directed operation EvaluateGeneratorBody takes arguments
+/// functionObject (an ECMAScript function object) and argumentsList (a List of
+/// ECMAScript language values) and returns a throw completion or a return
+/// completion.
+pub(crate) fn evaluate_generator_body(
+    agent: &mut Agent,
+    function_object: ECMAScriptFunction,
+    arguments_list: ArgumentsList,
+) -> JsResult<Value> {
+    // 1. Perform ? FunctionDeclarationInstantiation(functionObject, argumentsList).
+    function_declaration_instantiation(agent, function_object, arguments_list)?;
+
+    // 2. Let G be ? OrdinaryCreateFromConstructor(functionObject,
+    // "%GeneratorFunction.prototype.prototype%", « [[GeneratorState]],
+    // [[GeneratorContext]], [[GeneratorBrand]] »).
+    // 3. Set G.[[GeneratorBrand]] to empty.
+    let generator = ordinary_create_from_constructor(
+        agent,
+        function_object.into_function(),
+        ProtoIntrinsics::Generator,
+    )?;
+    let Object::Generator(generator) = generator else {
+        unreachable!()
+    };
+
+    // 4. Perform GeneratorStart(G, FunctionBody).
+    // SAFETY: We're alive so SourceCode must be too.
+    let body = unsafe {
+        agent[function_object]
+            .ecmascript_function
+            .ecmascript_code
+            .as_ref()
+    };
+    let is_concise_arrow_function = agent[function_object]
```

Comment:
**nitpick:** IIRC it's not possible for a concise arrow function to be a generator?

Reply:
True.